vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc

```diff
@@ -378,4 +378,210 @@ void FromStringToContributionList(
+ledger::MonthlyReportInfoPtr FromStringToMonthlyReport(
+    const std::string& data) {
+  base::Optional<base::Value> value = base::JSONReader::Read(data);
+
+  if (!value || !value->is_dict()) {
+    return nullptr;
+  }
+
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    return nullptr;
+  }
+
+  auto balance_report = ledger::BalanceReportInfo::New();
+  auto* balance = dictionary->FindDictKey("balance");
+  if (balance) {
+    const auto grants = balance->FindDoubleKey("grants");
+    if (grants) {
+      balance_report->grants = *grants;
+    }
+
+    const auto earning_from_ads = balance->FindDoubleKey("earning_from_ads");
+    if (earning_from_ads) {
+      balance_report->earning_from_ads = *earning_from_ads;
+    }
+
+    const auto auto_contribute = balance->FindDoubleKey("auto_contribute");
+    if (auto_contribute) {
+      balance_report->auto_contribute = *auto_contribute;
+    }
+
+    const auto recurring_donation =
+        balance->FindDoubleKey("recurring_donation");
+    if (recurring_donation) {
+      balance_report->recurring_donation = *recurring_donation;
+    }
+
+    const auto one_time_donation = balance->FindDoubleKey("one_time_donation");
+    if (one_time_donation) {
+      balance_report->one_time_donation = *one_time_donation;
+    }
+  }
+
+  ledger::TransactionReportInfoList transaction_report;
+  auto* transactions = dictionary->FindListKey("transactions");
+  if (transactions) {
+    for (auto& item : transactions->GetList()) {
+      auto transaction = ledger::TransactionReportInfo::New();
+
+      const auto amount = item.FindDoubleKey("amount");
+      if (amount) {
+        transaction->amount = *amount;
+      }
+
+      const auto type = item.FindIntKey("type");
+      if (type) {
+        transaction->type = static_cast<ledger::ReportType>(*type);
+      }
+
+      const auto* created_at = item.FindStringKey("created_at");
+      if (created_at) {
+        transaction->created_at = std::stoull(*created_at);
+      }
+
+      transaction_report.push_back(std::move(transaction));
+    }
+  }
+
+  ledger::ContributionReportInfoList contribution_report;
+  auto* contributions = dictionary->FindListKey("contributions");
+  if (contributions) {
+    for (auto& item : contributions->GetList()) {
+      auto contribution = ledger::ContributionReportInfo::New();
+
+      const auto* contribution_id = item.FindStringKey("contribution_id");
+      if (contribution_id) {
+        contribution->contribution_id = *contribution_id;
+      }
+
+      const auto amount = item.FindDoubleKey("amount");
+      if (amount) {
+        contribution->amount = *amount;
+      }
+
+      const auto type = item.FindIntKey("type");
+      if (type) {
+        contribution->type = static_cast<ledger::ReportType>(*type);
+      }
+
+      const auto* created_at = item.FindStringKey("created_at");
+      if (created_at) {
+        contribution->created_at = std::stoull(*created_at);
```

Comment:
Same: what happens if the string cannot be converted to uint64?

Reply:
added try catch

Reply:
FWIW, I was curious if there was something in chromium, and found `base::StringToUint64` [here](https://cs.chromium.org/chromium/src/base/strings/string_number_conversions.h?q=StringToUint64&sq=package:chromium&g=0&l=87) - not sure if it would work given that it takes a `StringPiece` though.